### PR TITLE
Optional user

### PR DESCRIPTION
**Pull request by ryanpbrewster**
The number of threads I ended up tugging on actually blew my mind.

@leogrim @andrewconner 

This compiles, but there is a 0% chance that it will pass tests. Tackling that next.

### CODE REVIEW

File: kifi-backend/modules/search/app/com/keepit/search/index/graph/keep/KeepIndexable.scala [hunk 1]
```diff
@@ -124,7 +124,7 @@ case class KeepIndexable(keep: Keep, sourceAttribution: Option[RawSourceAttribut
     }
 
     doc.add(buildIdValueField(uriIdField, keep.uriId))
-    doc.add(buildIdValueField(userIdField, keep.userId))
+    doc.add(buildIdValueField(userIdField, keep.userId.getOrElse(Id[User](-1)))) // TODO(ryan): defer to Léo on this, idk if this is right
```
Comment (ryanpbrewster) [inline kifi-backend/modules/search/app/com/keepit/search/index/graph/keep/KeepIndexable.scala:127]: @leogrim this is your party

**Comment by andrewconner** [comment]
I'm quite surprised the diff is this small.

File: kifi-backend/modules/common/app/com/keepit/model/Keep.scala [hunk 2]
```diff
@@ -250,7 +251,7 @@ object BasicKeep {
     (__ \ 'url).format[String] and
     (__ \ 'visibility).format[LibraryVisibility] and
     (__ \ 'libraryId).formatNullable[PublicId[Library]] and
-    (__ \ 'ownerId).format[ExternalId[User]] and
+    (__ \ 'ownerId).formatNullable[ExternalId[User]] and
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:254]: This breaks clients, yeah?

I may add a `deprecatedWrites` that will fill in a garbage UUID there.

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:254]: I don't know off the top of my head who uses `BasicKeep`, but yeah, don't remove any fields.

**Pull request closed without merging**